iOS/DuckDuckGo/AIChat/InputBox/SwitchBar/TextEntry/SwitchBarTextEntryViewController.swift

```diff
@@ -66,57 +74,63 @@ class SwitchBarTextEntryViewController: UIViewController {
         textEntryView.resignFirstResponder()
     }
 
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection) {
+            borderOverlayView.layer.borderColor = UIColor(Color(designSystemColor: .accent)).cgColor
+        }
+    }
+
     private func setupViews() {
         setupContainerViewAppearance()
+        setUpBorderOverlayAppearance()
+
         view.addSubview(containerView)
+        view.addSubview(borderOverlayView)
 
         containerView.addSubview(textEntryView)
-        containerView.backgroundColor = UIColor(designSystemColor: .surface)
 
         containerView.translatesAutoresizingMaskIntoConstraints = false
         textEntryView.translatesAutoresizingMaskIntoConstraints = false
+        borderOverlayView.translatesAutoresizingMaskIntoConstraints = false
     }
 
-    private func setupContainerViewAppearance() {
-
-        containerView.layer.cornerRadius = 16
-        containerView.layer.masksToBounds = false
-
-        containerView.layer.shadowColor = UIColor.label.cgColor
-        containerView.layer.shadowOffset = CGSize(width: 0, height: 2)
-        containerView.layer.shadowRadius = 8
-        containerView.layer.shadowOpacity = 0.1
-
-        containerView.layer.borderColor = UIColor(designSystemColor: .accent).cgColor // TODO: observe trait collection changes
-        containerView.layer.borderWidth = 2
+    private func setUpBorderOverlayAppearance() {
+        borderOverlayView.layer.cornerRadius = Metrics.borderCornerRadius
+        borderOverlayView.layer.masksToBounds = true
 
-        updateShadowPath()
+        borderOverlayView.layer.borderColor = UIColor(designSystemColor: .accent).cgColor
+        borderOverlayView.layer.borderWidth = Metrics.borderWidth
     }
 
-    private func updateShadowPath() {
-        containerView.layer.shadowPath = UIBezierPath(
-            roundedRect: containerView.bounds,
-            cornerRadius: containerView.layer.cornerRadius
-        ).cgPath
-    }
+    private func setupContainerViewAppearance() {
 
-    override func viewDidLayoutSubviews() {
-        super.viewDidLayoutSubviews()
-        updateShadowPath()
-    }
+        containerView.layer.cornerRadius = Metrics.containerCornerRadius
+        containerView.layer.masksToBounds = false
 
+        containerView.backgroundColor = UIColor(designSystemColor: .surface)
+        containerView.applyActiveShadow()
```

Comment:
This applies the correct shadow that's updated along with trait collection and size changes.